src/coreclr/jit/importercalls.cpp

```diff
@@ -9201,9 +9214,13 @@ void Compiler::impCheckCanInline(GenTreeCall*           call,
         CORINFO_CLASS_HANDLE clsHandle = compCompHnd->getMethodClass(ftn);
         unsigned const       clsAttr   = compCompHnd->getClassAttribs(clsHandle);
 
+        CallArg* pRetBuf = pParam->call->gtArgs.GetRetBufferArg();
+
         // Return type
         //
-        var_types const fncRetType = pParam->call->TypeGet();
+        var_types const fncRetType =
+            pRetBuf == nullptr ? pParam->call->TypeGet()
+                               : pParam->pThis->lvaGetRealType(pRetBuf->GetEarlyNode()->AsLclVarCommon()->GetLclNum());
```

Comment:
If anything this should use `gtRetClsHnd`.

Reply:
Is `gtReturnType` always available here?

Reply:
It seems that it's always available as we set it when created a Call node.

Reply:
Resolved by using `gtReturnType` instead.